python/ic_azure/azure_discover_dimensions.py

```diff
@@ -4,71 +4,100 @@
+class AzureDiscoverDimensions(object):
+    """Retrieve dimensions from Azure's resource"""
 
     def __init__(self, azure_client):
         self._client = azure_client.client()
         self.subscription_id = azure_client.subscription_id
         self.resources = azure_client.resources
 
-    # Method to retrieve roles from Azure's resource
-    def get_roles(self, resource, metric, dimension, metric_namespace):
-
-        # Declare variables
-        rolesList = []
+    def get_data(self, resource, metric, queryfilter, metric_namespace):
 
         # Calculate start/end times
-        end_time = datetime.utcnow() - timedelta(minutes=5)
+        end_time = datetime.now() - timedelta(minutes=5)
         start_time = end_time - timedelta(hours=1)
 
+        try:
+            # Retrieve instance data
+            metrics_data = self._client.metrics.list(
+                resource,
+                timespan="{}/{}".format(
+                    start_time.strftime('%Y-%m-%dT%H:%M:%SZ'),
+                    end_time.strftime('%Y-%m-%dT%H:%M:%SZ')
+                ),
+                interval="PT1H",
+                metricnames=metric,
+                aggregation="Total",
+                result_type="Metadata",
+                filter=queryfilter,
+                metricnamespace=metric_namespace
+            )
+        except Exception as error:
+            sys.exit(f"Client request failed. {error}")
+
+        return metrics_data
+
+    # Method to retrieve dimensions from Azure's resource
+    def get_dimensions(self, resource, metric, dimension, extent, metric_namespace):
+
+        # Declare variables
+        dimensionsList = []
+
         # Read resource from config using key
         if not resource.startswith("/subscriptions"):
             resource = self.resources.get(resource)
-
-        # Retrieve instance data
-        metrics_data = self._client.metrics.list(
-            resource,
-            timespan="{}/{}".format(
-                start_time.strftime('%Y-%m-%dT%H:%M:%SZ'),
-                end_time.strftime('%Y-%m-%dT%H:%M:%SZ')
-            ),
-            interval="PT1H",
-            metricnames=metric,
-            aggregation="Total",
-            result_type="Metadata",
-            filter=dimension + " eq '*'",
-            metricnamespace=metric_namespace
-        )
+        
+        filter = dimension + " eq '*'"
+        result = self.get_data(resource, metric, filter, metric_namespace)
 
         # Loop through metric data and retrieve instances
-        for item in metrics_data.value:
+        for item in result.value:
             for timeserie in item.timeseries:
                 for data in timeserie.metadatavalues:
-
-                    # Don't add duplicates into roles list
-                    if data.__dict__.get("value") in rolesList:
+                    # Don't add duplicates into dimensions list
+                    name = data.__dict__.get("value")
+                    if name in dimensionsList:
```

Comment:
I don't think this duplicate check works as is now that `dimensionsList` is an array of objects and not an array of dimension names. It probably should instead check if there is an object in `dimensionsList` array where `"{#DIMENSION}"` property value equals `name`.